storage/txcache/txListBySenderMap.go

```diff
@@ -102,24 +102,64 @@ func (txMap *txListBySenderMap) RemoveSendersBulk(senders []string) uint32 {
 	return nRemoved
 }
 
-// GetListsSortedByOrderNumber gets the list of sender addreses, sorted by the global order number
+type txListBySenderSortKind string
+
+// LRUCache is currently the only supported Cache type
+const (
+	SortByOrderNumberAsc txListBySenderSortKind = "SortByOrderNumberAsc"
+	SortBySmartScoreAsc  txListBySenderSortKind = "SortBySmartScoreAsc"
+)
+
+func (txMap *txListBySenderMap) GetListsSortedBy(sortKind txListBySenderSortKind) []*txListForSender {
+	switch sortKind {
+	case SortByOrderNumberAsc:
+		return txMap.GetListsSortedByOrderNumber()
+	case SortBySmartScoreAsc:
+		return txMap.GetListsSortedBySmartScore()
+	default:
+		return txMap.GetListsSortedBySmartScore()
```

Comment:
sort should be by gasPrice.

Reply:
Removed sorting overall. We do sort at insertion time for senders as well.